### PR TITLE
feat: add ARM node provider

### PR DESCRIPTION
**Pull request by JesusAlvarezTorres**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
/kind feature

**What this PR does / why we need it**:
Adds ARM node provider for custom clouds where IMDS is not available. Primary use case for this change is to support Azure Stack Hub.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:
Adding ARM node provider required for Azure Stack clouds where IMDS compute is not yet available.

**Release note**:
```
feat: add ARM node provider 

```

### CODE REVIEW

**Comment by JesusAlvarezTorres** [comment]
> Thanks for your pull request. Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).
> 
> 📝 **Please follow instructions at https://git.k8s.io/community/CLA.md#the-contributor-license-agreement to sign the CLA.**
> 
> It may take a couple minutes for the CLA signature to be fully registered; after that, please reply here with a new comment and we'll verify. Thanks.
> 
> * If you've already signed a CLA, it's possible we don't have your GitHub username or you're using a different email address.  Check your existing CLA data and verify that your [email is set on your git commits](https://help.github.com/articles/setting-your-email-in-git/).
> * If you signed the CLA as a corporation, please sign in with your organization's credentials at https://identity.linuxfoundation.org/projects/cncf to be authorized.
> * If you have done the above and are still having issues with the CLA being reported as unsigned, please log a ticket with the Linux Foundation Helpdesk: https://support.linuxfoundation.org/
> * Should you encounter any issues with the Linux Foundation Helpdesk, send a message to the backup e-mail support address at: [login-issues@jira.linuxfoundation.org](mailto:login-issues@jira.linuxfoundation.org)

Already signed the CLA

**Comment by nilo19** [comment]
this change makes pkg/node/node.go useless right? why not deleting it?

**Comment by nilo19** [comment]
@JesusAlvarezTorres could you please fix the test failure?

File: cmd/cloud-node-manager/app/config/config.go [hunk 1]
```diff
@@ -32,6 +32,12 @@ type Config struct {
 	// the node name
 	NodeName string
 
+	// Specifies if node information is retrieved via IMDS or ARM.
+	UseInstanceMetadata bool
```
Comment (andyzhangx) [inline cmd/cloud-node-manager/app/config/config.go:36]: there is already `"useInstanceMetadata": true`, shall we leverage that directly?
https://github.com/kubernetes-sigs/cloud-provider-azure/blob/master/examples/azure.json#L27

Comment (JesusAlvarezTorres) [reply cmd/cloud-node-manager/app/config/config.go]: I don't think we can, this flag is set at the time of cloud creation; we needed an additional flag to tell the cloud creation to use the azure.json config file or not. Thus I added the UseInstanceMetadata flag at the command level to guide the code to use the config file or not.

**Comment by nilo19** [comment]
@JesusAlvarezTorres could you squash the commits?

File: pkg/node/nodearm.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
+/*
+Copyright 2019 The Kubernetes Authors.
```
Comment (feiskyer) [inline pkg/node/nodearm.go:2]: s/2019/2021

File: pkg/node/nodeproviderfactory.go [hunk 3]
```diff
@@ -0,0 +1,34 @@
+/*
+Copyright 2019 The Kubernetes Authors.
```
Comment (feiskyer) [inline pkg/node/nodeproviderfactory.go:2]: s/2019/2021

**Review by feiskyer — APPROVED** [review]
looks good in general. Could you squash the commits?

**Comment by JesusAlvarezTorres** [comment]
> @JesusAlvarezTorres could you squash the commits?

done, squashed commits

File: pkg/node/nodeproviderfactory.go (hunk 3 above)
Comment (JesusAlvarezTorres) [reply pkg/node/nodeproviderfactory.go]: change made

File: pkg/node/nodearm.go (hunk 2 above)
Comment (JesusAlvarezTorres) [reply pkg/node/nodearm.go]: change made

**Comment by JesusAlvarezTorres** [comment]
> looks good in general. Could you squash the commits?

done, squashed the commits. Thanks for the review.

**Pull request merged**